src/main/kotlin/com/example/ramenbm/domain/user/service/impl/UserAuthServiceImpl.kt

```diff
@@ -1,39 +1,57 @@
 package com.example.ramenbm.domain.user.service.impl
 
-import com.example.ramenbm.domain.user.exception.DuplicateEmailException
-import com.example.ramenbm.domain.user.exception.PasswordNotCorrectException
-import com.example.ramenbm.domain.user.exception.UserNotFoundException
+import com.example.ramenbm.domain.user.exception.*
 import com.example.ramenbm.domain.user.presentation.data.dto.request.SignInRequest
 import com.example.ramenbm.domain.user.presentation.data.dto.request.SignUpRequest
 import com.example.ramenbm.domain.user.presentation.data.dto.request.toEntity
-import com.example.ramenbm.domain.user.presentation.data.dto.response.SignInResponse
+import com.example.ramenbm.domain.user.presentation.data.dto.response.TokenResponse
+import com.example.ramenbm.domain.user.repository.RefreshTokenRepository
 import com.example.ramenbm.domain.user.repository.UserRepository
 import com.example.ramenbm.domain.user.service.UserAuthService
-import com.example.ramenbm.global.annotation.ServiceWithTransaction
+import com.example.ramenbm.global.security.jwt.TokenParser
 import com.example.ramenbm.global.security.jwt.TokenProvider
+import org.springframework.data.repository.findByIdOrNull
 import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
 
-@ServiceWithTransaction
+@Service
 class UserAuthServiceImpl(
-        private val userRepository: UserRepository,
-        private val passwordEncoder: PasswordEncoder,
-        private val tokenProvider: TokenProvider
-): UserAuthService {
+    private val userRepository: UserRepository,
+    private val refreshTokenRepository: RefreshTokenRepository,
+    private val passwordEncoder: PasswordEncoder,
+    private val tokenProvider: TokenProvider,
+    private val tokenParser: TokenParser
+) : UserAuthService {
```

Comment:
전체적인 코드 컨벤션을 보니까 (): 상위 클래스 이런식으로 작성 하셨는데 컨변션을 지키면 좋을거 같습니다!
```suggestion
): UserAuthService {
```

Reply:
수정했습니다.